src/coreclr/debug/ee/debuggermessagebox.cpp

```diff
@@ -0,0 +1,174 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//*****************************************************************************
+// DebuggerMessageBox.cpp
+//
+//*****************************************************************************
+
+#include "stdafx.h"                     // Standard header.
+#include <utilcode.h>                   // Utility helpers.
+#include <corerror.h>
+#include <clrversion.h>
+#include "../../dlls/mscorrc/resource.h"
+
+// Output printf-style formatted text to the debugger if it's present or stdout otherwise.
+static void DbgPrintf(const LPCSTR szFormat, ...)
+{
+    char szBuffer[4096];
+
+    va_list args;
+    va_start(args, szFormat);
+
+    _vsnprintf_s(szBuffer, ARRAY_SIZE(szBuffer), _TRUNCATE, szFormat, args);
+
+    va_end(args);
+
+    if (IsDebuggerPresent())
+    {
+        OutputDebugStringUtf8(szBuffer);
+    }
+    else
+    {
+        fprintf(stdout, "%s", szBuffer);
+        fflush(stdout);
+    }
+}
+
+typedef int (*MessageBoxWFnPtr)(HWND hWnd,
+                                LPCWSTR lpText,
+                                LPCWSTR lpCaption,
+                                UINT uType);
+
+static int MessageBoxImpl(
+                  LPCWSTR title,        // Dialog box title
+                  LPCWSTR message,      // Message
+                  UINT uType)
+{
+    CONTRACTL
+    {
+        // May pump messages.  Callers should be GC_TRIGGERS and MODE_PREEMPTIVE,
+        // but we can't include EE contracts here.
```

Comment:
Can we include the `STATIC_CONTRACT_*` contract variants that are no-ops at runtime? I know they don't do anything, but they're still good for documentation purposes and useful if we ever bring back our static analysis contract validation tools from the dead.

Reply:
This is from the original implementation. I didn't change anything about the contracts.